### DISCUSSION TITLE
get too many 'Connection reset by peer' error in the monitored port logs

### DISCUSSION THREAD

**Discussion by Mr-YYM** [megaease/easeprobe#556] [General]
被检测端收到许多 Connection reset by peer 报错，我知道这是监控主动发出 RST 强制断开 TCP 链接导致的，为了防止 TIME_WAIT 过多的问题。但是我的程序不想收到此类报错，希望能够新增一个开关来使用正常的四次挥手关闭连接。

The monitored side is receiving many 'Connection reset by peer' errors. I understand that this is caused by the monitoring tool actively sending RST packets to forcibly close the TCP connection in order to prevent an excessive number of TIME_WAIT states. However, I don't want my program to receive such errors. I would like to add an option to use the normal four-way handshake to close the connection instead.

<img width="847" alt="image" src="https://github.com/user-attachments/assets/2a3b32db-f725-4e58-baaf-3835c6293e56">

**Comment by proditis** [comment]
I like that. Maybe we can introduce an option, for the affected probes, to let the users choose if they want to use `SetLinger` or not?

**Reply by Mr-YYM** [reply]
I agree that! Do you have some time to implement it? ❤️

**Reply by proditis** [reply]
A PR with an initial attempt to implement this has been opened but it needs testing to ensure it doesnt break things. Unfortunately my Go knowledge is limited so i hope this works :D

**Reply by Mr-YYM** [reply]
Thank you for your great work. 👍
